jdaviz/configs/specviz/helper.py

```diff
@@ -280,26 +280,33 @@ def get_data(self, data_label=None, cls=None, subset_to_apply=None):
         ----------
         data_label : str, optional
             Provide a label to retrieve a specific data set from data_collection.
+        spectral_subset : str, optional
+            Spectral subset applied to data.
         cls : `~specutils.Spectrum1D`, optional
             The type that data will be returned as.
-        subset_to_apply : str, optional
-            Subset that is to be applied to data before it is returned.
 
         Returns
         -------
         data : cls
             Data is returned as type cls with subsets applied.
 
         """
+        spatial_subset = kwargs.pop("spatial_subset", None)
+        function = kwargs.pop("function", None)
```

Comment:
If not cubeviz, we still want the kwargs error to be raised for these two

```suggestion
        if self.app.config == 'cubeviz':
            spatial_subset = kwargs.pop("spatial_subset", None)
            function = kwargs.pop("function", None)
```

Reply:
I don't think I understand, why do you want to move these lines inside the if statement? Doing that leaves the `specivz.get_data` call open to some of the misuse we talked about last week, i.e. `specviz.get_data(data, spctrl_subset='1')`

Reply:
the kwargs check will still be outside the if-statement, but then for non-cubeviz, passing `spatial_subset` and/or `function` will trigger that check and raise an error.  In short, I think `specviz.get_data(data, spatial_subset='Subset 1')` _should_ raise an error.

Reply:
I added an `elif` after this `if` statement that raises a `ValueError`.

Reply:
ah, I see, you assume both variables exist to pass them on later in either case.  Then I think the `else` needs to also set `spatial_subset = None`.